app/services/discovery_report.rb

```diff
@@ -19,6 +19,35 @@ def initialize(batch)
     @summary = { objects_with_error: 0, mimetypes: Hash.new(0), start_time: Time.now.utc.to_s, total_size: 0 }
   end
 
+  # @return [String] output_path to store report results, generate a different string each time
+  def output_path
+    Dir::Tmpname.create(["#{self.class.name.underscore}_", '.json'], batch.batch_context.output_dir) { |path| path }
+  end
+
+  # @return [Boolean]
+  def using_file_manifest?
+    using_file_manifest && File.exist?(File.join(bundle_dir, batch.batch_context.file_manifest))
+  end
+
+  # @return [PreAssembly::FileManifest]
+  def file_manifest
+    @file_manifest ||= PreAssembly::FileManifest.new(csv_filename: batch.batch_context.file_manifest, bundle_dir: bundle_dir)
+  end
+
+  # By using jbuilder on an enumerator, we reduce memory footprint (vs. to_a)
+  # @return [Jbuilder] (caller needs obj.to_builder.target! for the JSON string)
+  def to_builder
+    json_report = Jbuilder.new do |json|
+      json.rows { json.array!(each_row) }
+      json.summary summary.merge(end_time: Time.now.utc.to_s)
+    end
+    @objects_had_errors = (summary[:objects_with_error] > 0) # indicate if any objects generated errors
+    @error_message = "#{summary[:objects_with_error]} objects had errors in the discovery report" if objects_had_errors
+    json_report
+  end
+
+  private
```

Comment:
the private keyword is added here, and i don't see it removed elsewhere in the diff, so i think that indicates a lack of any private methods in this class before this PR (as opposed to counter-intuitive diff presentation)